src/main/java/com/splunk/hecclient/HecAckPoller.java

```diff
@@ -184,6 +184,41 @@ public int getAckPollInterval() {
         return ackPollInterval;
     }
 
+    public void stickySessionHandler(HecChannel channel) {
+        // Set channel unavailable
+        channel.setAvailable(false);
+        log.debug("Channel {} set to be not available", channel.getId());
+        // Get batches for the channel
+        ConcurrentHashMap<Long, EventBatch> channelBatches = outstandingEventBatches.get(channel);
+        // Remove batches for the channel from the poller
+        if(channelBatches != null && channelBatches.size() > 0) {
+            totalOutstandingEventBatches.addAndGet(-channelBatches.size());
+            // Remove batches from kafka record tracker to fail them and resend
+            if (pollerCallback != null) {
+                List<EventBatch> expired = new ArrayList<>();
+                Iterator iter = channelBatches.entrySet().iterator();
+                while(iter.hasNext()) {
+                    Map.Entry<Long, EventBatch> pair = (Map.Entry) iter.next();
+                    EventBatch batch = pair.getValue();
+                    batch.fail();
+                    expired.add(batch);
+                    iter.remove();
+                }
```

Comment:
include log lines indicated   - Found x batches marked for failure 
                                               - Sucessfully failed x batches